calculator.rb

```diff
@@ -0,0 +1,106 @@
+# acceptable operators
+#this is inelegant I know, I ran out of time!
+operators = [ "\+","\-","\*","\/","divide","DIVIDE","Divide","Div","DIV",
+  "div","add","addition","Add","ADD","ADDITION","Addition",
+  "subtract","Subtract","sub","Sub","SUB","SUBTRACT","mult",
+  "multiply","Multiply","MULTIPLY"
+]
+
+# methods
+def add(num_one, num_two)
+  return convert_input(num_one) + convert_input(num_two)
+
+end
+
+def subtract(num_one, num_two)
+  return convert_input(num_one) - convert_input(num_two)
+
+end
+
+def multiply(num_one, num_two)
+  return convert_input(num_one) * nconvert_input(num_two)
+
+end
+
+def divide(num_one, num_two)
+  return convert_input(num_one)/ convert_input(num_two)
+
+end
+# need to change the reg exp to reject things like punctuation
+def num_check(first_num)
+
+  while first_num =~ /[a-zA-Z]/
+    print "That is not a valid number. Please Enter a number."
+    first_num = gets.chomp
```

Comment:
This method will check that the number is valid, and prompt the user to try again if it's not, but you don't have any way to access the final number outside this method. Remember that the parameter `first_num` is a only a copy of the number from outside.

If you want to get the final number back, you'll have to return it:

```ruby
def num_check(first_num)
  while first_num =~ /[a-zA-Z]/
    print "That is not a valid number. Please Enter a number."
    first_num = gets.chomp
  end
  return first_num
end
```

And then when calling the method below, keep track of the returned value:

```ruby
print "\nenter a number\s"
first_num = gets.chomp

# checks if user input is a number
first_num = num_check(first_num)
```